pkg/api/validation/validation.go

```diff
@@ -461,6 +466,28 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 	return allErrs
 }
 
+func ValidateNoNewFinalizers(newFinalizers []string, oldFinalizers []string, fldPath *field.Path) field.ErrorList {
+	const newFinalizersErrorMsg string = `no new finalizers can be added if the object is being deleted`
+	allErrs := field.ErrorList{}
+	// we expect number of finalizers to be small, otherwise we should construct a set to optimize.
```

Comment:
Hm, I guess I would check if newFinalizers is non-zero and then use this one-liner if it is:

```
extra := sets.NewString(newFinalizers...).Subtract(sets.NewString(oldFinalizers))
```

up to you.